packages/web-components/src/button/button.styles.ts

```diff
@@ -142,10 +142,14 @@ export const baseButtonStyles = css`
     margin-inline-end: var(--icon-spacing);
   }
 
-  :is([slot='end'], ::slotted([slot='end'])) {
+  :host(:not(${iconOnlyState})) :is([slot='end'], :host(:not(${iconOnlyState}))::slotted([slot='end'])) {
     margin-inline-start: var(--icon-spacing);
   }
 
+  slot {
```

Comment:
Is this needed? It seems a bit dangerous to set `<slot>` to have `display` other than `contents`.